src/model.js

```diff
@@ -1058,37 +1058,48 @@ var Model = TransisObject.extend(function() {
   };
 
   // Public: Runs registered validators for the given attribute. This will clear any existing
-  // validation errors for the given attribute.
+  // validation errors for the given attribute. The registered validators can be reduced by
+  // providing an optional context.
   //
   // name - The name of the attribute to run validations for.
+  // ctx  - The context in which to run validations on the given attribute. (optional)
   //
   // Returns `true` if no validation errors are found on the given attribute and `false` otherwise.
-  this.prototype.validateAttr = function(name) {
+  this.prototype.validateAttr = function(name, ctx) {
     this._clearErrors(name);
-
     if (!this.validators[name]) { return true; }
 
-    for (let i = 0, n = this.validators[name].length; i < n; i++) {
-      let validator = this.validators[name][i];
-      if (typeof validator === 'function') { validator.call(this); }
-      else if (typeof validator === 'string' && validator in this) { this[validator](); }
-      else { throw new Error(`${this.constructor}#validateAttr: don't know how to execute validator: \`${validator}\``); }
+    let attrCtx = this.validators[name].find( (v)=> {return v.on && v.on === ctx} );
```

Comment:
Small Per Issuef: We are decreasing the performance of majority case (no context) by traversing the validators twice (here and in the `else`).
We could easily short circuit this first traversal by checking whether we have a `ctx` first, so...

```javascript
let attrCtx = ctx && this.validators[name].find( (v)=> {return v.on && v.on === ctx} );
```